orchestra/migrations/0087_todo_task_data_migration.py

```diff
@@ -0,0 +1,28 @@
+# Generated by Django 2.2.13 on 2020-09-04 14:49
+
+from django.db import migrations
+from django.db.models import OuterRef
+from django.db.models import Subquery
+
+
+def fill_project_and_step_fields(apps, schema_editor):
+    Step = apps.get_model('orchestra', 'Step')
+    Project = apps.get_model('orchestra', 'Project')
+    Todo = apps.get_model('orchestra', 'Todo')
+    o = OuterRef('id')
+    step_outerref_filter_qs = Step.objects.filter(tasks__todos__id=o)
+    project_outterref_filter_qs = Project.objects.filter(tasks__todos__id=o)
+    subq_step = step_outerref_filter_qs.values('id')[:1]
+    subq_proj = project_outterref_filter_qs.values('id')[:1]
+    Todo.objects.update(project=Subquery(subq_proj), step=Subquery(subq_step))
```

Comment:
having some sort of testing/verification might be nice. like just at the shell, before the code runs, loop through all todos and show that the number of todos where the task.project == project and task.step == step is 0, and after the migration, the number that meet that criterion should be the number of todos.

Reply:
(not code reviewed code, just verification pre-/post-migration.

Reply:
This generates the following SQL–snippet:
```sql
 UPDATE "orchestra_todo"
SET    "project_id" =
       (
                  SELECT     u0."id"
                  FROM       "orchestra_project" U0
                  INNER JOIN "orchestra_task" U1
                  ON         (
                                        u0."id" = u1."project_id")
                  INNER JOIN "orchestra_todo" U2
                  ON         (
                                        u1."id" = u2."task_id")
                  WHERE      u2."id" = ("orchestra_todo"."id") limit 1),
       "step_id" =
       (
                  SELECT     u0."id"
                  FROM       "orchestra_step" u0
                  INNER JOIN "orchestra_task" u1
                  ON         (
                                        u0."id" = u1."step_id")
                  INNER JOIN "orchestra_todo" u2
                  ON         (
                                        u1."id" = u2."task_id")
                  WHERE      u2."id" = ("orchestra_todo"."id") limit 1)
WHERE  "orchestra_todo"."is_deleted" = false;args=(false,) 
```

Reply:
I intuitively tried to apply `Todo.objects.update(project=F('task__project'), step=F('task__step'))`, but it raises a Field error. But "joined F() aren't supported by design" according to this ticket: https://code.djangoproject.com/ticket/14104